File: Java/src/de/tom/fibonacciNumbers/FibonacciNumbers.java
package de.tom.fibonacciNumbers;

import java.util.Scanner;

public class FibonacciNumbers {


	public static void main(String[] args) {

		int number = 0;

		Scanner scanner = new Scanner(System.in);

		System.out.println("Please choose a not negativ integer number: ");
		number = scanner.nextInt();

		System.out.println("The "+number+" number of the fibinacci numbers is "+ generateFibonacciNumber(number));
		System.out.println("The "+number+" number of the fibinacci numbers is "+ generateExplicidFibonacci(number));
		System.out.println("The aproximation on the golden ratio is " + goldenRatio(number));
		scanner.close();

	}

	private static long generateFibonacciNumber(int numberOfFibonacciSeries) {
		if (numberOfFibonacciSeries == 0 || numberOfFibonacciSeries == 1) return numberOfFibonacciSeries;

		return generateFibonacciNumber(numberOfFibonacciSeries - 1) + generateFibonacciNumber(numberOfFibonacciSeries - 2);
	}

	private static double generateExplicidFibonacci(int numberOfFibonacciSeries) {

		double x = 1 / Math.sqrt(5);
		double y = Math.pow(((1 + Math.sqrt(5)) / 2), numberOfFibonacciSeries);
		double z = Math.pow(((1 - Math.sqrt(5)) / 2), numberOfFibonacciSeries);

		return x * (y - z);

	}

	private static double goldenRatio(int numberOfFibonacciSeries) {

		return (1.0*generateExplicidFibonacci(numberOfFibonacciSeries)) / generateExplicidFibonacci(numberOfFibonacciSeries-1);

	}

}

File: C++/FibonacciNumbers.cpp
#include "FibonacciNumbers.h"
#include <stdio.h>
#include <Math.h>

int main() {

    int number = 0;

    printf("Please choose a not negativ integer number\n");

    scanf_s("%d", &number);

    printf("The %d number of the fibinacci numbers is %d\n", number, generateExplicidFibonacci(number));
    printf("The %d number of the fibinacci numbers is %d\n", number, generateFibonacci(number));
    
    return 0;

}

long long generateExplicidFibonacci(int numberOfFibonacciSeries) {

    double x = 1 / sqrt(5);
    double y = pow(((1 + sqrt(5)) / 2), numberOfFibonacciSeries);
    double z = pow(((1 - sqrt(5)) / 2), numberOfFibonacciSeries);

    return x * (y - z);

}

long long generateFibonacci(int numberOfFibonacciSeries) {

    if (numberOfFibonacciSeries == 0 || numberOfFibonacciSeries == 1) return numberOfFibonacciSeries;

    return generateFibonacci(numberOfFibonacciSeries - 1) + generateFibonacci(numberOfFibonacciSeries - 2);

}
